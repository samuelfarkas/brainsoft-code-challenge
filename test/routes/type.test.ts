import { build } from '../helper'
const app = build()

describe('type endpoints', () => {
  it('should return all types', async () => {
    const res = await app.inject({
      url: '/types'
    })
    const payload = res.json()
    expect(res.statusCode).toEqual(200)
    expect(payload).toHaveLength(18)
    expect(payload).toMatchSnapshot()
  })
})
