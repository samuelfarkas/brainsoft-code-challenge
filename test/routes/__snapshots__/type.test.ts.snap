// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type endpoints should return all types 1`] = `
[
  {
    "id": 1,
    "name": "Grass",
  },
  {
    "id": 2,
    "name": "Poison",
  },
  {
    "id": 3,
    "name": "Normal",
  },
  {
    "id": 4,
    "name": "Water",
  },
  {
    "id": 5,
    "name": "Electric",
  },
  {
    "id": 6,
    "name": "Fighting",
  },
  {
    "id": 7,
    "name": "Fairy",
  },
  {
    "id": 8,
    "name": "Fire",
  },
  {
    "id": 9,
    "name": "Ice",
  },
  {
    "id": 10,
    "name": "Bug",
  },
  {
    "id": 11,
    "name": "Dragon",
  },
  {
    "id": 12,
    "name": "Flying",
  },
  {
    "id": 13,
    "name": "Psychic",
  },
  {
    "id": 14,
    "name": "Steel",
  },
  {
    "id": 15,
    "name": "Ground",
  },
  {
    "id": 16,
    "name": "Rock",
  },
  {
    "id": 17,
    "name": "Dark",
  },
  {
    "id": 18,
    "name": "Ghost",
  },
]
`;
